code/modules/mob/living/simple_animal/guardian/guardian.dm

```diff
@@ -0,0 +1,526 @@
+	projectiletype = /obj/item/projectile/neurotox
+	ranged_cooldown_cap = 1
+	projectilesound = 'sound/weapons/pierce.ogg'
+	ranged = 1
+	rapid = 1
+	range = 13
+	playstyle_string = "As a ranged type, you have only light damage resistance, but are capable of spraying neurotoxin."
+	magic_fluff_string = "..And draw the Sentinel, an alien master of ranged combat."
+	tech_fluff_string = "Boot sequence complete. Ranged combat modules active. Nanoswarm online."
+	bio_fluff_string = "Your scarab swarm finishes mutating and stirs to life, capable of spitting neurotoxin."
+
+
+/mob/living/simple_animal/hostile/guardian/bluespace
+	ranged = 1
+	range = 15
+	melee_damage_lower = 15
+	melee_damage_upper = 15
+	speed = -1
+	attack_sound = 'sound/weapons/emitter.ogg'
+	projectiletype = /obj/item/projectile/magic/teleport
+	projectilesound = 'sound/weapons/emitter.ogg'
+	playstyle_string = "As a bluespace type, you have only light damage resistance, but are capable of shooting teleporation bolts as well as flinging enemies away with your standard attack."
+	magic_fluff_string = "..And draw the Wizard, master of teleportation."
+	tech_fluff_string = "Boot sequence complete. Experimental bluespace combat modules active. Nanoswarm online."
+	bio_fluff_string = "Your scarab swarm finishes mutating and stirs to life, crackling with bluespace energy."
+
+/mob/living/simple_animal/hostile/guardian/bluespace/AttackingTarget()
+	..()
+	if(istype(target, /atom/movable))
+		var/atom/movable/M = target
+		if(!M.anchored && M != src.summoner)
+			do_teleport(M, M, 10)
+
+/mob/living/simple_animal/hostile/guardian/bomb
+	melee_damage_lower = 15
+	melee_damage_upper = 15
+	damage_transfer = 0.6
+	range = 13
+	playstyle_string = "As an explosive type, you have only moderate close combat abilities, but are capable of converting any adjacent item into a disguised bomb via shift click."
+	magic_fluff_string = "..And draw the Scientist, master of explosive death."
+	tech_fluff_string = "Boot sequence complete. Explosive modules active. Nanoswarm online."
+	bio_fluff_string = "Your scarab swarm finishes mutating and stirs to life, capable of stealthily booby trapping items."
+	var/bomb_cooldown = 0
+
+/mob/living/simple_animal/hostile/guardian/bomb/ShiftClickOn(atom/movable/A)
+	if(istype(A, /obj/))
+		if(bomb_cooldown <= world.time && !stat)
+			var/obj/item/weapon/guardian_bomb/B = new /obj/item/weapon/guardian_bomb(get_turf(A))
+			src << "<span class='danger'><B>Success! Bomb armed!</span></B>"
+			bomb_cooldown = world.time + 400
+			B.spawner = src
+			B.disguise (A)
+		else
+			src << "<span class='danger'><B>Your powers are on cooldown! You must wait 40 seconds between bombs.</span></B>"
+
+/obj/item/weapon/guardian_bomb
+	name = "bomb"
+	desc = "You shouldn't be seeing this!"
+	var/obj/stored_obj
+	var/mob/living/spawner
+
+
+
+/obj/item/weapon/guardian_bomb/proc/disguise(var/obj/A)
+	A.loc = src
+	stored_obj = A
+	name = A.name
+	desc = A.desc
+	icon = A.icon
```

Comment:
replace:
icon
icon_state
overlays
and I believe the pixel_X/Ys

with appearance = A.appearance